lib/base.js

```diff
@@ -201,7 +201,7 @@ class TCPBase extends Base {
       endTime: -1,
     };
     let endTime;
-    this._socket.write(packet.data, () => {
+    meta.succ = this._socket.write(packet.data, () => {
```

Comment:
Returns true if the entire data was flushed successfully to the kernel buffer. Returns false if all or part of the data was queued in user memory. 'drain' will be emitted when the buffer is again free.